src/cli.ts

```diff
@@ -4,22 +4,45 @@ import {cli, define} from 'gunshi';
 import * as prompts from '@clack/prompts';
 import c from 'picocolors';
 import {report} from './index.js';
-import type {Message, PackType} from './types.js';
+import type {PackType} from './types.js';
+import {LocalDependencyAnalyzer} from './analyze-dependencies.js';
+import { pino } from 'pino';
 
 const version = createRequire(import.meta.url)('../package.json').version;
 const allowedPackTypes: PackType[] = ['auto', 'npm', 'yarn', 'pnpm', 'bun'];
 
+// Create a logger instance with pretty printing for development
+const logger = pino({
+  transport: {
+    target: 'pino-pretty',
+    options: {
+      colorize: true,
+      translateTime: 'SYS:standard',
+      ignore: 'pid,hostname',
+    },
+  },
+});
+
 const defaultCommand = define({
   options: {
     pack: {
       type: 'string',
       default: 'auto',
       description: `Package manager to use for packing ('auto' | 'npm' | 'yarn' | 'pnpm' | 'bun')`
+    },
+    'log-level': {
+      type: 'string',
+      default: 'info',
+      description: 'Set the log level (debug | info | warn | error)'
     }
   },
   async run(ctx) {
     const root = ctx.positionals[0];
     let pack = ctx.values.pack as PackType;
+    const logLevel = ctx.values['log-level'] as string;
```

Comment:
i would've expected gunshi to have already typed this as a string. what does your editor say it is without the cast? out of interest

Reply:
I had a red squiggly before, it's gonne i removed the type cast.